.github/workflows/publish.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish thundra-demo-todo-app-java to Docker Hub
+
+on:
+  pull_request:
+    branches: [ master ]
+    types: [ closed ]
+
+jobs:
+  docker:
+    name: Publish - Docker Hub
+    if: github.event.pull_request.merged == true
```

Comment:
If I am not mistaken, this provides to publish only it is merged. Why on-push event can be used for it instead of both of them?